docs/move_clusters.md

```diff
@@ -0,0 +1,59 @@
+# Moving Hive-managed Clusters
+
+## Summary
+
+There is a need to be able to move Hive-managed clusters from one Hive instance to another.
+This may be useful if the user want to replace a Hive instance with a new one or if the user
+wants to implement sharding between Hive instances.
+
+Hive will manage all of the moving for the user. The user will create a CR indicating with
+details about the destination cluster and label selectors to match against the cluster to
+move.
+
+## ClusterRelocator CRD
+
+A ClusterRelocator is cluster-scoped.
+
+This is an example of a ClusterRelocator that may be used to move all of the ClusterDeployments
+with the label `hive-instance: hive2` to the remote cluster accessed via the kubeconfig secret
+at "hive2-remote-kubecofig-secret" in the "hive" namespace.
+
+```yaml
+apiVersion: hive.openshift.io/v1
+kind: ClusterRelocator
+metadata:
+  name: hive2-cluster-relocator
+spec:
+  kubeconfigSecretRef:
+    name: hive2-remote-kubeconfig-secret
+    namespace: hive
+  clusterDeploymentSelector:
+    matchLabels:
+      hive-instance: hive2
+```
+
+## Process
+
+There is a controller that will reconcile ClusterDeployments and watch ClusterDeployments
+and ClustereRelocators. For ClusterDeployments that match with a ClusterRelocator, the
+controller will move the ClusterDeployment to the Hive instance associated with the kubeconfig
+referenced in the ClusterRelocator.
+
+1. Add the `hive.openshift.io/relocating=<relocator-name>` annotation to the ClusterDeployment. This informs other controllers not to take action on the cluster.
+This will block syncsets and machinesets from being applied and will block the clusterdeployment
+controller from running its finalizer code for the ClusterDeployment.
```

Comment:
Will the annotation be applied to a clusterdeployment in any state? or will we wait for an ongoing provision/deprovision to finish before applying the annotation?

Reply:
If a provision is ongoing, then the clusterdeployment controller will stop the provisioning when relocation commences.

If deprovisioning is ongoing, then the ClusterDeployment is not eligible to be relocated. We should not relocate deleted resources.